libs/state/src/lib/convert-partials-to.observable.ts

```diff
@@ -0,0 +1,40 @@
+import { combineLatest, from, isObservable, Observable, startWith } from 'rxjs';
+import { Injector, isSignal, Signal } from '@angular/core';
+import { toObservable } from '@angular/core/rxjs-interop';
+import { map } from 'rxjs/operators';
+import { coalesceWith } from '@rx-angular/cdk/coalescing';
+import { Promise } from '@rx-angular/cdk/zone-less/browser';
+
+export function convertPartialsToObservable<
+  T extends object,
+  K extends keyof T
+>(
+  input: Partial<{ [K: string]: Observable<T[K]> | Signal<T[K]> }>,
+  injector?: Injector
+): Observable<Partial<T>> {
+  const keys = Object.keys(input);
+
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-return
```

Comment:
I would get rid of this comment and probably throw like so:

 ```ts
 const observables = keys.map((key) => {
    const obsOrSignal = input[key];

    if (isObservable(obsOrSignal)) {
      return obsOrSignal?.pipe(startWith(undefined));
    }
    if (isSignal(obsOrSignal)) {
      return toObservable(obsOrSignal, { injector });
    }

    throw new TypeError(`Expected type Observable or Signal for key "${key}"`);
  });
  ```
  
  Wdyt?

Reply:
Like!

Reply:
done